jdaviz/configs/imviz/tests/test_catalogs.py

```diff
@@ -99,3 +101,47 @@ def test_plugin_image_with_result(self, imviz_helper):
         catalogs_plugin.vue_do_clear()
 
         assert not catalogs_plugin.results_available
+
+        # test loading from file
+        table = imviz_helper.app._catalog_source_table
+        skycoord_table = SkyCoord(table['ra'],
+                                  table['dec'],
+                                  unit='deg')
+        qtable = QTable({'sky_centroid': skycoord_table})
+        tmp_file = tmp_path / 'test.ecsv'
+        qtable.write(tmp_file, overwrite=True)
+
+        catalogs_plugin.from_file = str(tmp_file)
+        # setting filename from API will automatically set catalog to 'From File...'
+        assert catalogs_plugin.catalog.selected == 'From File...'
+        catalogs_plugin.vue_do_search()
+        assert catalogs_plugin.results_available
+        assert catalogs_plugin.number_of_results == 2473
+
+    def test_from_file_parsing(self, imviz_helper, tmp_path):
```

Comment:
This test should be taken out of the test class because this one should not need remote data.